AberrantSMPP/SMPPCommunicator.cs

```diff
@@ -622,7 +622,10 @@ public SmppResponse SendRequest(SmppRequest request)
 			lock (_RequestsAwaitingResponse)
 			{
 				state = new RequestState(SendPdu(request));
-				_RequestsAwaitingResponse.Add(state.SequenceNumber, state);
+				if (!_RequestsAwaitingResponse.ContainsKey(state.SequenceNumber))
```

Comment:
I am afraid this is still racy.. We will need to use some kind of concurrent dictionary or wrap this contains+add logic within a lock() {} statement.

Reply:
Hi! Why is it not accepted to master?